templates.rst

```diff
@@ -0,0 +1,1061 @@
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                https://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <route id="blog_index"
+                path="/"
+                controller="App\Controller\BlogController::index"/>
+
+            <route id="blog_post"
+                path="/article/{slug}"
+                controller="App\Controller\BlogController::show"/>
+        </routes>
+
+    .. code-block:: php
+
+        // config/routes.php
+        use App\Controller\BlogController;
+        use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->add('blog_index', '/')
+                ->controller([BlogController::class, 'index'])
+            ;
+
+            $routes->add('blog_post', '/articles/{slug}')
+                ->controller([BlogController::class, 'show'])
+            ;
+        };
+
+Use the ``path()`` Twig function to link to these pages and pass the route name
+as the first argument and the route parameters as the optional second argument:
+
+.. code-block:: html+twig
+
+    <a href="{{ path('blog_index') }}">Homepage</a>
+
+    {# ... #}
+
+    {% for post in blog_posts %}
+        <h1>
+            <a href="{{ path('blog_post', {slug: post.slug}) }}">{{ post.title }}</a>
+        </h1>
+
+        <p>{{ post.excerpt }}</p>
+    {% endfor %}
+
+The ``path()`` function generates relative URLs. If you need to generate
+absolute URLs (for example when rendering templates for emails or RSS feeds),
+use the ``url()`` function, which takes the same arguments as ``path()``
+(e.g. ``<a href="{{ url('blog_index') }}"> ... </a>``).
+
+.. _templates-link-to-assets:
+
+Linking to CSS, JavaScript and Image Assets
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Templates usually link to images, JavaScript, stylesheets and other assets.
+Instead of writing the asset URLs by hand, use the ``asset()`` function to
+generate the URLs.
+
+To use this function, install the *asset* package:
+
+.. code-block:: terminal
+
+    $ composer require symfony/asset
+
+You can now use the ``asset()`` function:
+
+.. code-block:: html+twig
+
+    <img src="{{ asset('images/logo.png') }}" alt="Symfony!"/>
+
+    <link href="{{ asset('css/blog.css') }}" rel="stylesheet"/>
+
+    <script src="{{ asset('bundles/acme/js/loader.js') }}"></script>
+
+The ``asset()`` function's main purpose is to make your application more portable.
+If your application lives at the root of your host (e.g. ``https://example.com``),
+then the rendered paths should be ``/images/logo.png``. But if your application
+lives in a subdirectory (e.g. ``https://example.com/my_app``), each asset path
+should render with the subdirectory (e.g. ``/my_app/images/logo.png``). The
```

Comment:
```suggestion
will render with the subdirectory (e.g. ``/my_app/images/logo.png``). The
```

Reply:
Because of how the phrase is redacted, I think `should` is more correct. The phrase structure is: *"if the app is in the root dir, assets should [...] but if the app is in a subdir, assets should [...] The assets() function takes care of this."*